packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -630,7 +637,13 @@ abstract class DatabaseInstanceNew extends DatabaseInstanceBase implements IData
   protected enableIamAuthentication?: boolean;
 
   constructor(scope: Construct, id: string, props: DatabaseInstanceNewProps) {
-    super(scope, id);
+    // RDS always lower-cases the ID of the database
+    const dbInstanceId = Token.isUnresolved(props.instanceIdentifier)
+      ? props.instanceIdentifier
+      : props.instanceIdentifier?.toLowerCase();
```

Comment:
I can't find equivalent unit tests? I would expect to see a test case that says 'dbInstanceIdentifier is lower case' or something.

Reply:
> feels a bit odd here.
> 
> Wouldn't this mean, to a new user, we're giving the impression that upper case character is allowed, but actually we're modifying it under the hood?
> It could get confusing such that the user is looking for the db with upper case chars but don't find it, either manual search on the console or programmatically in the cdk or somewhere else.
> 
> Would it be better to simply make this a validation and tell the customer to provide lower case?

I don't think so. CloudFormation allows passing the uppercase version, and it gets lowercased on the server side, and I don't see why would we diverge from this behavior.

Anyway, this PR is not for re-hashing this decision, which has already been made (and released to customers) in https://github.com/aws/aws-cdk/pull/13710. This PR is only for adding support for cross-environment references to Instance. 

I've added a comment clarifying that, and also changed the title of the PR.

Reply:
> I can't find equivalent unit tests? I would expect to see a test case that says 'dbInstanceIdentifier is lower case' or something.

A test like this [is here](https://github.com/aws/aws-cdk/blob/fef8c802de2a7bd56ff75ba059cc4054b6d2e318/packages/%40aws-cdk/aws-rds/test/instance.test.ts#L1283-L1305).